packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -97,15 +101,19 @@ class NetworkAssetBundle extends AssetBundle {
   @override
   Future<ByteData> load(String key) async {
     final http.Response response = await _httpClient.get(_urlFromKey(key));
-    if (response.statusCode == 200)
-      return null;
+    if (response.statusCode != 200)
+      throw new FlutterError('Unable to load asset: $key');
     return response.bodyBytes.buffer.asByteData();
   }
 
   @override
   Future<String> loadString(String key, { bool cache: true }) async {
     final http.Response response = await _httpClient.get(_urlFromKey(key));
-    return response.statusCode == 200 ? response.body : null;
+    if (response.statusCode != 200)
+      throw new FlutterError(
+          'Unable to load asset: $key\n'
+          'HTTP status code: ${response.statusCode}');
```

Comment:
nit: braces around multiline statements, and if you put a newline after the (, put a newline before the ).